### PR TITLE
[all] Add cursor rules to CSDK

### PR DESCRIPTION
**Pull request by MenKNas**
- [ ] This PR follows the [Contribution Guide](https://github.com/Sitecore/content-sdk/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

### Add Cursor AI coding agent rules for consistent development patterns

This PR implements Cursor rules to guide AI-assisted development across the Sitecore Content SDK repository, ensuring consistent code quality and adherence to established patterns.

What's Added:

- Core rule set (.cursor/rules/) with 9 focused guidance files covering general principles, JavaScript/TypeScript standards, and Sitecore-specific patterns
- Documentation updates to guide developers on AI-assisted workflows

Key Benefits:

- AI assistants now enforce Content SDK naming conventions and best practices
- Automatic generation of proper JSDoc following existing codebase patterns
- Consistent Sitecore component development with field validation and error handling
- Zero runtime impact - documentation-only guidance files
- Backward compatibility and following Cursor's best practices

## Testing Details
- [ ] Unit Test Added
- [x] Manual Test/Other => Conducted comprehensive validation including syntax checking of all .mdc files, verification of glob pattern file targeting, compliance with Cursor's 500-line rule limit, and practical testing by creating a sample Sitecore component that successfully demonstrated AI guidance following all established naming conventions, error handling patterns, and JSDoc standards. All rules provide actionable guidance when using Cursor AI in the repository.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: .cursor/rules/general.mdc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+Version Control:
+
+- Write descriptive commit messages
+- Keep commits focused and atomic
+- Use branching strategies appropriate to team size
+- Review code before merging
+
+Documentation:
+
+- Document public APIs and interfaces
```
Comment (art-alexeyenko) [inline .cursor/rules/general.mdc:65]: for these rules, do we need to have them specified explicitly, considering we're using jsdoc only within monorepo?

File: .cursor/rules/general.mdc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+## Quality Assurance
+
+Code Review:
+
+- Review for logic, readability, and maintainability
+- Check error handling and edge cases
+- Verify tests cover new functionality
+- Ensure documentation is updated
+
+Continuous Integration:
```
Comment (art-alexeyenko) [inline .cursor/rules/general.mdc:86]: Same question for this. Is there an expectation for cursor to handle CI?

File: .cursor/rules/project-context.mdc [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+description: Project-wide context and constraints for create-content-sdk-app
+alwaysApply: true
+globs: []
+---
+
+# Project Context
+
+Purpose: Scaffolds example apps for the Sitecore Content SDK, providing templates and a CLI initializer.
```
Comment (art-alexeyenko) [inline .cursor/rules/project-context.mdc:9]: Should this and other files be located in the `create-content-sdk` root?

File: .cursor/rules/project-context.mdc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Language: TypeScript (Node 22+)
```
Comment (art-alexeyenko) [inline .cursor/rules/project-context.mdc]: Would it make a difference if we mention Node as LTS versions here and at other places? Saves us trouble of updating the version in Cursor file every year

File: .cursor/rules/sitecore.mdc [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+// Good: Using Sitecore field components
+<Text field={fields?.title} tag="h1" />
+<RichText field={fields?.content} />
+
+// Avoid: Manual field value extraction unless necessary
+```
+
+Layout Service:
+
+- Use the Layout Service for page data fetching
```
Comment (art-alexeyenko) [inline .cursor/rules/sitecore.mdc]: Perhaps we should instruct it to prefer using our scClient methods, with potential for expansion

File: .cursor/rules/project-context.mdc (hunk 3 above)
Comment (MenKNas) [reply .cursor/rules/project-context.mdc:9]: Good catch, this is the first place we should add the rules for our users. I have added them within the templates (as this is what users actually get, so I think this is the right place to be). My recommendation would be to keep the initial rules as well at the repo root level, so team members and, of course, external contributors can benefit from those. We can discuss this further.

File: .cursor/rules/general.mdc (hunk 1 above)
Comment (MenKNas) [reply .cursor/rules/general.mdc:65]: This is related to monorepo rules so it's about imposing them on that level. It's optional for the users to have those rules in the app.

File: .cursor/rules/general.mdc (hunk 2 above)
Comment (MenKNas) [reply .cursor/rules/general.mdc:86]: In this case, it’s more about providing context (not required, but it doesn’t hurt either)

File: .cursor/rules/project-context.mdc (hunk 4 above)
Comment (MenKNas) [reply .cursor/rules/project-context.mdc]: I think that's a solid point, will update to support LTS versions.

File: .cursor/rules/repo-structure.mdc [hunk 6]
```diff
@@ -0,0 +1,32 @@
+---
+description: Repository structure and templates policy
+alwaysApply: true
+globs: []
+---
+
+# Repository Structure
+
+/src
```
Comment (art-alexeyenko) [inline .cursor/rules/repo-structure.mdc]: let's add a path `packages/create-content-sdk` at the top here for clarity

**Review by art-alexeyenko — APPROVED** [review]
